src/agent/src/image.rs

```diff
@@ -0,0 +1,329 @@
+                    "Image bundle config path: {:?}", image_oci_config_path
+                );
+
+                let image_oci =
+                    oci::Spec::load(image_oci_config_path.to_str().ok_or_else(|| {
+                        anyhow!(
+                            "Invalid container image OCI config path {:?}",
+                            image_oci_config_path
+                        )
+                    })?)
+                    .context("load image bundle")?;
+
+                if let (Some(container_root), Some(image_root)) =
+                    (container_oci.root.as_mut(), image_oci.root.as_ref())
+                {
+                    let root_path = Path::new(CONTAINER_BASE)
+                        .join(container_id)
+                        .join(image_root.path.clone());
+                    container_root.path = String::from(root_path.to_str().ok_or_else(|| {
+                        anyhow!("Invalid container image root path {:?}", root_path)
+                    })?);
+                }
+
+                if let (Some(container_process), Some(image_process)) =
+                    (container_oci.process.as_mut(), image_oci.process.as_ref())
+                {
+                    self.merge_oci_process(container_process, image_process);
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Partially merge an OCI process specification into another one.
+    fn merge_oci_process(&self, target: &mut oci::Process, source: &oci::Process) {
+        // Override the target args only when the target args is empty and source.args is not empty
+        if target.args.is_empty() && !source.args.is_empty() {
+            target.args.append(&mut source.args.clone());
+        }
+
+        // Override the target cwd only when the target cwd is blank and source.cwd is not blank
+        if target.cwd == "/" && source.cwd != "/" {
+            target.cwd = String::from(&source.cwd);
+        }
+
+        for source_env in &source.env {
+            if let Some((variable_name, variable_value)) = source_env.split_once('=') {
+                debug!(
+                    sl(),
+                    "source spec environment variable: {variable_name:?} : {variable_value:?}"
+                );
+                if !target.env.iter().any(|i| i.contains(variable_name)) {
+                    target.env.push(source_env.to_string());
+                }
+            }
+        }
+    }
+}
+#[cfg(test)]
+mod tests {
+    use super::ImageService;
+    use rstest::rstest;
+
+    #[rstest]
+    // Image cwd should override blank container cwd
+    // TODO - how can we tell the user didn't specifically set it to `/` vs not setting at all? Is that scenario valid?
+    #[case("/", "/imageDir", "/imageDir")]
+    // Container cwd should override image cwd
+    #[case("/containerDir", "/imageDir", "/containerDir")]
+    // Container cwd should override blank image cwd
+    #[case("/containerDir", "/", "/containerDir")]
```

Comment:
Now we are using rstest, I wonder if, rather than comments, it might be cleaner to use case descriptions: https://docs.rs/rstest/latest/rstest/attr.rstest.html#optional-case-description?

Reply:
Sure.

Reply:
Done. Thanks @stevenhorsman